plugin/file/secondary.go

```diff
@@ -160,23 +165,32 @@ Restart:
 
 		case <-refreshTicker.C:
 
+			start = time.Now()
+			RefreshSeconds.WithLabelValues(z.origin).Observe(time.Since(start).Seconds())
+
 			time.Sleep(jitter(5000)) // 5s randomize
 
 			ok, err := z.shouldTransfer()
 			if err != nil {
+				FailedRefreshCount.WithLabelValues(z.origin).Inc()
 				log.Warningf("Failed refresh check %s", err)
 				retryActive = true
 				continue
 			}
 
 			if ok {
+				UpdatedSerial.WithLabelValues(z.origin).Set(float64(z.Apex.SOA.Serial))
 				if err := z.TransferIn(); err != nil {
 					// transfer failed
+
+					FailedRefreshCount.WithLabelValues(z.origin).Inc()
 					retryActive = true
 					break
 				}
+				UpdatedSuccess.WithLabelValues(z.origin).Set(float64(time.Now().Unix()))
 			}
 
+			RefreshCount.WithLabelValues(z.origin).Inc()
```

Comment:
Isn't refreshcount just updateSuccess+failedRefreshcount? If so, it can be removed.

Reply:
No, `updateSuccess` is the timestamp of the last valid update. This looks correct.